spec/integration/apply_compile_spec.rb

```diff
@@ -376,7 +376,7 @@ def get_notifies(result, future = false)
       it 'serializes Error objects in apply blocks' do
         result = run_cli_json(%w[plan run puppet_types::error] + config_flags)
         notify = get_notifies(result, true)
-        expect(notify[0]['title']).to eq("ApplyResult resource: The command failed with exit code 1")
+        expect(notify[0]['title']).to eq("ApplyResult resource: The command failed with exit code 127")
```

Comment:
I don't understand why this changed?

Reply:
It changed because we now wrap the command in `sh -c`. If you instruct `sudo` to run a non-existent command, it will fail with exit code 1. But if you wrap it in `sh`, then `sudo` successfully runs the shell, which _then_ can't find the command and exits with the standard 127.